File: [C]hashmap/[C]hashmap/242-valid-anagram.java
class Solution {
  public boolean isAnagram(String s, String t) {
      if (s.length() != t.length()) {
          return false;
      }
      HashMap<Character, Integer> sCharToFreq = new HashMap<>();
      HashMap<Character, Integer> tCharToFreq = new HashMap<>();
      for (int i = 0; i < s.length(); i++) {
          Character sc = s.charAt(i);
          sCharToFreq.compute(sc, (k, v) -> v == null ? 1 : v + 1);
          Character tc = t.charAt(i);
          tCharToFreq.compute(tc, (k, v) -> v == null ? 1 : v + 1);
      }
      for (Character c: sCharToFreq.keySet()) {
          if (!sCharToFreq.get(c).equals(tCharToFreq.get(c))) {
              return false;
          }
      }
      return true;
  }
}

// time O(n)
// space O(26) == O(1), constant
// using hashmap and store sth’s freq

File: [C]hashmap/[C]hashmap/242-valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) != len(t):
            return False
        schar_freq = [0 for _ in range(26)]
        tchar_freq = [0 for _ in range(26)]
        for c in s:
            schar_freq[ord(c) - ord('a')] += 1
        for c in t:
            tchar_freq[ord(c) - ord('a')] += 1
        for i in range(26):
            if schar_freq[i] != tchar_freq[i]:
                return False
        return True

# time O(n)
# space O(26) == O(1), constant
# using hashmap and store sth’s freq